File: Models/Stics_soil_temperature/src/java/Stics_soil_temperature/SoiltempState.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;
public class SoiltempState
{
    private Double [] prev_temp_profile;
    private Double temp_amp;
    private Double therm_amp;
    private Double [] temp_profile;
    
    public SoiltempState() { }
    
    public SoiltempState(SoiltempState toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            prev_temp_profile = new Double[1];
        for (int i = 0; i < 1; i++)
        {
            prev_temp_profile[i] = toCopy.getprev_temp_profile()[i];
        }
            this.temp_amp = toCopy.gettemp_amp();
            this.therm_amp = toCopy.gettherm_amp();
            temp_profile = new Double[toCopy.gettemp_profile().length];
        for (int i = 0; i < toCopy.gettemp_profile().length; i++)
        {
            temp_profile[i] = toCopy.gettemp_profile()[i];
        }
        }
    }
    public Double [] getprev_temp_profile()
    { return prev_temp_profile; }

    public void setprev_temp_profile(Double [] _prev_temp_profile)
    { this.prev_temp_profile= _prev_temp_profile; } 
    
    public Double gettemp_amp()
    { return temp_amp; }

    public void settemp_amp(Double _temp_amp)
    { this.temp_amp= _temp_amp; } 
    
    public Double gettherm_amp()
    { return therm_amp; }

    public void settherm_amp(Double _therm_amp)
    { this.therm_amp= _therm_amp; } 
    
    public Double [] gettemp_profile()
    { return temp_profile; }

    public void settemp_profile(Double [] _temp_profile)
    { this.temp_profile= _temp_profile; } 
    
}

File: Models/Stics_soil_temperature/src/cs/Stics_soil_temperature/soil_tempState.cs
using System;
using System.Collections.Generic;
public class Soil_tempState 
{
    private double[] _prev_temp_profile;
    private double _prev_canopy_temp;
    private double _temp_amp;
    private double[] _temp_profile;
    private double[] _layer_temp;
    private double _canopy_temp_avg;
    
    /// <summary>
    /// Constructor of the soil_tempState component")
    /// </summary>  
    public soil_tempState() { }
    
    
    public soil_tempState(soil_tempState toCopy, bool copyAll) // copy constructor 
    {
        if (copyAll)
        {
    
            prev_temp_profile = new double[toCopy.prev_temp_profile.Length];
            for (int i = 0; i < toCopy.prev_temp_profile.Length; i++)
                { prev_temp_profile[i] = toCopy.prev_temp_profile[i]; }
    
            prev_canopy_temp = toCopy.prev_canopy_temp;
            temp_amp = toCopy.temp_amp;
            temp_profile = new double[toCopy.temp_profile.Length];
            for (int i = 0; i < toCopy.temp_profile.Length; i++)
                { temp_profile[i] = toCopy.temp_profile[i]; }
    
            layer_temp = new double[toCopy.layer_temp.Length];
            for (int i = 0; i < toCopy.layer_temp.Length; i++)
                { layer_temp[i] = toCopy.layer_temp[i]; }
    
            canopy_temp_avg = toCopy.canopy_temp_avg;
        }
    }
    public double[] prev_temp_profile
    {
        get { return this._prev_temp_profile; }
        set { this._prev_temp_profile= value; } 
    }
    public double prev_canopy_temp
    {
        get { return this._prev_canopy_temp; }
        set { this._prev_canopy_temp= value; } 
    }
    public double temp_amp
    {
        get { return this._temp_amp; }
        set { this._temp_amp= value; } 
    }
    public double[] temp_profile
    {
        get { return this._temp_profile; }
        set { this._temp_profile= value; } 
    }
    public double[] layer_temp
    {
        get { return this._layer_temp; }
        set { this._layer_temp= value; } 
    }
    public double canopy_temp_avg
    {
        get { return this._canopy_temp_avg; }
        set { this._canopy_temp_avg= value; } 
    }
}